homeassistant/components/http/ban.py

```diff
@@ -249,8 +282,28 @@ def _add_ban(self, ip_ban: IpBan) -> None:
             # Write in a single write call to avoid interleaved writes
             out.write("\n" + yaml.dump(ip_))
 
+    def _write_bans(self, ip_bans: dict[IPv4Address | IPv6Address, IpBan]) -> None:
```

Comment:
I'm sure this was considered, but it looks like `async_add_ban` adds to the memory dict, and then calls `_add_ban` which appends the address while and the new `_write_bans` method writes the entire file. This means that `_write_bans`, which sounds pretty generic, is really only used in one circumstance.

It probably doesn't make sense to use this for `add` because append should be more efficient, but probably worth a comment.

Reply:
True, append is more efficient then writing the whole file over and over again.